services/viva/microservice/src/lambdas/decideCaseStatus.ts

```diff
@@ -60,11 +63,16 @@ export async function decideCaseStatus(
   const { caseKeys, workflow } = input.detail;
 
   const newStatusType = decideNewCaseStatus(workflow);
-  if (newStatusType == undefined) {
+  const newState = calculateNewState(workflow);
+
+  const isUndefined = newStatusType == undefined || newState == undefined;
+  if (isUndefined) {
```

Comment:
What exactly is undefined? Perhaps a more explaining variable name would be better?